main/guides/zoe/proposal.md

```diff
@@ -4,69 +4,85 @@
 
 ## Making An offer
 
-To make an offer, you use [`E(zoe).offer()`](/reference/zoe-api/zoe.md#e-zoe-offer-invitation-proposal-paymentkeywordrecord-offerargs), which takes three arguments:
+To make an offer, you use [`E(zoe).offer(...)`](/reference/zoe-api/zoe.md#e-zoe-offer-invitation-proposal-paymentkeywordrecord-offerargs), which takes up to four arguments:
 - An **invitation** to participate in this contract instance.
 - A **proposal** stating your offer conditions.
-- The **payments** escrowed for the offer, each in association with a `proposal`-specified keyword.
+- The **payments** escrowed for the offer, each corresponding with a **give** [Keyword](/reference/zoe-api/zoe-data-types.md#keyword) in **proposal**.
+- **offerArgs** expressing additional arguments for the **offerHandler** associated with the invitation by [`zcf.makeInvitation(...)`](/reference/zoe-api/zoe-contract-facet.md#zcf-makeinvitation-offerhandler-description-customproperties-proposalshape).
 
 ## Invitations
 
-`Invitations` are a special case of ERTP `payments`. They are linked to a specific contract `instance`, and
-having one gives you the right to participate in that contract `instance`, for example, by making offers in it.
+An [Invitation](/reference/zoe-api/zoe-data-types.md#invitation) is a special case of ERTP [Payment](/reference/ertp-api/payment.md). Each is linked to a specific contract [Instance](/reference/zoe-api/zoe-data-types.md#instance), and
+having one gives you the right to participate in that contract instance by making offers in it.
 
-There are two main ways for contract users to get an `invitation`:
-- If you create the contract `instance`, you get a special creator `invitation`.
-- Someone (possibly you) who holds the right objects has created an `invitation` for a contract `instance` and gives it to
+There are two main ways for contract users to get an invitation:
```

Comment:
Does the generality of the second bullet cover everything not covered by the first bullet? I think so. If so, then these are not just the "two main ways", since that phrase implies that some case isn't covered.